templates/todo/api/csharp-cosmos-sql-func/ListsFunctions.cs

```diff
@@ -0,0 +1,241 @@
+
+        await _repository.UpdateList(existingList);
+
+        return response;
+        
+    }
+
+
+    [Function("DeleteList")]
+    public async Task<HttpResponseData> DeleteList(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "delete", Route = "lists/{list_id}")]
+        HttpRequestData req, string list_id)
+    {
+        var response = req.CreateResponse(HttpStatusCode.NoContent);
+        response.Headers.Add("Content-Type", "application/json; charset=utf-8");
+        if (await _repository.GetListAsync(list_id) == null)
+        {
+            return req.CreateResponse(HttpStatusCode.NotFound); ;
+        }
+        await _repository.DeleteListAsync(list_id);
+        return response;
+    }
+
+    [Function("GetListItems")]
+    public async Task<HttpResponseData> GetListItems(
+        [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "lists/{list_id}/items/{skip:int?}/{batchSize:int?}")]
+        HttpRequestData req, string list_id, int? skip, int? batchSize)
+    {
+        var response = req.CreateResponse(HttpStatusCode.OK);
+        response.Headers.Add("Content-Type", "application/json; charset=utf-8");
+
+        if (await _repository.GetListAsync(list_id) == null)
+        {
+            return req.CreateResponse(HttpStatusCode.NotFound); 
+        }
+        var items = await _repository.GetListItemsAsync(list_id, skip, batchSize);
+        string jsonString = JsonSerializer.Serialize(items);
+        response.WriteString(jsonString);
+        return response;
+    }
+
+
+    [Function("CreateListItem")]
+    public async Task<HttpResponseData> CreateListItem(
+           [HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = "lists/{list_id}/items")] HttpRequestData req, string list_id, string name, string? state, string? description)
+    {
+        var response = req.CreateResponse(HttpStatusCode.Created);
+        response.Headers.Add("Content-Type", "application/json; charset=utf-8");
+
+        if(await _repository.GetListAsync(list_id) == null)
+        {
+            return req.CreateResponse(HttpStatusCode.NotFound); 
+        }
+        
+    
+
+        var newItem = new TodoItem(list_id, name)
+        {
+            Name = name,
+            Description = description,
+            State = (state == null ? "todo": state),
+            CreatedDate = DateTimeOffset.UtcNow
+        };
+
+        await _repository.AddListItemAsync(newItem);
+        string jsonString = JsonSerializer.Serialize(newItem);
+        response.WriteString(jsonString);
+        
+        return response;
+    }
+
+
+    [Function("GetListItem")]
+    public async Task <HttpResponseData> GetListItem(
+        [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "lists/{list_id}/items/{item_id}")] HttpRequestData req,
+        string item_id, string list_id)
+    {
+        var response = req.CreateResponse(HttpStatusCode.OK);
+        response.Headers.Add("Content-Type", "application/json; charset=utf-8");
```

Comment:
We may want to wrap the calls to the REST api or use a lib to help us.

Reply:
I think there are libs or the GetJson.... method on some .NET object.  Can't recall what it is right now.

Reply:
Could you clarify this for me? Is this so we wouldn't have to create a response in each function?

Reply:
Like this: https://github.com/jongio/memealyzer/blob/main/src/net/WebApp/ServiceClient/ApiServiceClient.cs#L17  GetFromJsonAsync, I believe, sets content type so you don't have to here.

Reply:
Ok, in latest commit I added the "customHeaders" value for contentType in host.json and turns out that lets you set global behavior for HTTP functions.